src/main/java/edu/agh/wfiis/designpatterns/patternsUsageExample/DesignPatternsManager.java

```diff
@@ -1,47 +1,53 @@
 package edu.agh.wfiis.designpatterns.patternsUsageExample;
 
+import java.util.*;
+import java.io.*;
 
-import java.io.File;
-
-public class DesignPatternsManager {
-
-    public DesignPattern create(PatternName patternName) {
-        switch (patternName) {
-            case BUILDER: {
-                DesignPattern pattern = new DesignPattern("1","builder","builder pattern",null, true, false, false);
-                return pattern;
-            }
-            case DECORATOR: {
-                DesignPattern pattern = new DesignPattern("2","decorator","decorator pattern",null, true, false, false);
-                return pattern;
-            }
-            case STRATEGY: {
-                DesignPattern pattern = new DesignPattern("3","strategy","strategy pattern",null, false, false, true);
-                return pattern;
-            }
-            default: {
-                DesignPattern pattern = new UndefinedPattern();
-                return pattern;
-            }
+public class DesignPatternsManager implements Serializable {
+
+
+    private static volatile DesignPatternsManager instance;
+    private Map<String, DesignPattern> patterns = new HashMap<String, DesignPattern>();
+
+    public DesignPattern create(String patternName) {
+         return patterns.get(patternName);
+    }
+
+    public static DesignPatternsManager getInstance() {
+        if (instance == null) {
+            instance = new DesignPatternsManager();
         }
+        return instance;
+    }
+
+    private DesignPatternsManager() {
+        if (instance != null) {
+            throw new IllegalStateException("Cannot create new instance, please use getInstance method instead.");
+        }
+        patterns.put("BUILDER", DesignPattern.builder("1", "builder").withDescription("builder pattern").withExample(null).withPatternCode(null).build());
+        patterns.put("DECORATOR", DesignPattern.builder("2", "decorator").withDescription("decorator pattern").withExample(null).withPatternCode(null).build());
```

Comment:
Jeśli nie chcecie podawać example i patternCode, to nie używajcie metod withExample i withPatternCode. Nie są przecież obowiązkowe, a wygląda to trochę dziwnie.
Dodatkowo można to jeszcze ładnie sformatować.